File: spring/src/main/java/com/moseoh/assistant/response/ErrorResponse.java
package com.moseoh.assistant.response;

import java.time.LocalDateTime;

import com.moseoh.assistant.utils.exception.ServiceException.ErrorCode;

import org.springframework.http.ResponseEntity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
@AllArgsConstructor
public class ErrorResponse {
    private final LocalDateTime timestamp;
    private final int status;
    private final String error;
    private final String errorCode;
    private final String message;

    public static ResponseEntity<ErrorResponse> toResponseEntity(ErrorCode errorCode) {
        return ResponseEntity.status(errorCode.getHttpStatus())
                .body(ErrorResponse.builder()
                        .timestamp(LocalDateTime.now())
                        .status(errorCode.getHttpStatus().value())
                        .error(errorCode.getHttpStatus().name())
                        .errorCode(errorCode.name())
                        .message(errorCode.getDetail())
                        .build());
    }
}

File: flutter/lib/api/response/error_response.dart
class ErrorResponse {
  DateTime timestamp;
  int status;
  String error;
  String errorCode;
  String message;

  ErrorResponse({
    required this.timestamp,
    required this.status,
    required this.error,
    required this.errorCode,
    required this.message,
  });

  ErrorResponse.fromJson(Map json)
      : timestamp = DateTime.parse(json['timestamp']),
        status = json['status'],
        error = json['error'],
        errorCode = json['errorCode'],
        message = json['message'];
}
